pkg/operator/status.go

```diff
@@ -31,12 +31,8 @@ func (optr *Operator) syncAvailableStatus() error {
 	// set available
 	cov1helpers.SetStatusCondition(&co.Status.Conditions, configv1.ClusterOperatorStatusCondition{
 		Type: configv1.OperatorAvailable, Status: configv1.ConditionTrue,
-		Message: fmt.Sprintf("Cluster is available at %s", optrVersion),
+		Message: fmt.Sprintf("Cluster has deployed %s", optrVersion),
 	})
-	// clear progressing
```

Comment:
Why this drop?

Reply:
setting Available doesn't necessarily clear progressing (taken from docs https://github.com/openshift/installer/blob/master/docs/user/overview.md)
```
Failing is true with a detailed message Unable to apply 4.0.1: could not update 0000_70_network_deployment.yaml because the resource type NetworkConfig has not been installed on the server.
Available is true with message Cluster has deployed 4.0.0
Progressing is true with message Unable to apply 4.0.1: a required object is missing
```

It also looks like something other operators are doing (setting Avaiable can still be true while Failing and Progressing are also true)

Reply:
The other scenarios I've seen in other operators set the Conditions indipendently from the other conditions as well (which makes sense I believe)

Reply:
> setting Available doesn't necessarily clear progressing (taken from docs https://github.com/openshift/installer/blob/master/docs/user/overview.md)
> 
> ```
> Failing is true with a detailed message Unable to apply 4.0.1: could not update 0000_70_network_deployment.yaml because the resource type NetworkConfig has not been installed on the server.
> Available is true with message Cluster has deployed 4.0.0
> Progressing is true with message Unable to apply 4.0.1: a required object is missing
> ```
> 
> It also looks like something other operators are doing (setting Avaiable can still be true while Failing and Progressing are also true)

you are quoting an example when operator is progressing or failing...
when the operator is available
```
```

Reply:
I should check the other Conditions I guess right

Reply:
But still, why there's an example of all three to True then? Thats really confusing.

Reply:
updated

Reply:
anyway, the status you quoted can be already converge to without my latest additions (though they're indeed needed). Progressing and Failing are set _before_ available anyway so when we're at Avaiable, we know where we are (I think)

Reply:
fixed here I believe now https://github.com/openshift/machine-config-operator/pull/406/files#diff-bee1f8f36240cb95db10244fdf335146R37